unitex/src/fr/umlv/unitex/graphtools/FindAndReplace.java

```diff
@@ -396,4 +415,212 @@ public static String checkReplaceAll(ArrayList<GenericGraphBox> boxes, String se
+        graphicalZone.setHighlight(genericGraphBox, true);
+      }
+      return true;
+    }
+    if (boxEquals(findSeqList.get(0), genericGraphBox.getContent(), caseSensitive, useRegex) && genericGraphBox
+      .getTransitions().size() == 1) {
+      if (highlight) {
+        //genericGraphBox.setSelected(highlight);
+        genericGraphBox.setHighlight(true);
+        graphicalZone.getSelectedBoxes().add(genericGraphBox);
+        graphicalZone.setHighlight(genericGraphBox, true);
+      }
+      if (!findNextInSeq(graphicalZone, findSeqList, genericGraphBox.getTransitions().get(0), 1, highlight,
+        caseSensitive, useRegex)) {
+        graphicalZone.unSelectAllBoxes();
+        graphicalZone.setHighlight(false);
+        graphicalZone.removeHighlight();
+        return false;
+      }
+      return true;
+    }
+    graphicalZone.setHighlight(false);
+    graphicalZone.unSelectAllBoxes();
+    graphicalZone.removeHighlight();
+    return false;
+  }
+
+  /**
+   * Replace all sequence with replaceSeqList and returns the number of sequences
+   * which have been replaced.
+   *
+   * @param zone the GraphicalZone containing the sequence.
+   * @param findSeqList the list containing the sequence to search for.
+   * @param replaceSeqList the list containing the sequence to replace with.
+   * @param caseSensitive true if the search must be case sensitive, false otherwise.
+   * @param useRegex true if the search must use regular expressions, false otherwise.
+     * @return the number of sequences which have been replaced.
+     */
+  public static int replaceAllSeq(GraphicalZone zone, ArrayList<String> findSeqList, ArrayList<String>
+    replaceSeqList, boolean caseSensitive, boolean useRegex) {
+    ArrayList<GenericGraphBox> boxes = new ArrayList<GenericGraphBox>();
+    for (GenericGraphBox box : zone.getBoxes()) {
+      if (isSeq(zone, findSeqList, box, false, caseSensitive, useRegex)) {
+        boxes.add(box);
+      }
+    }
+    int res = 0;
+    for (GenericGraphBox box : boxes) {
+      isSeq(zone, findSeqList, box, true, caseSensitive, useRegex);
+      ArrayList<GenericGraphBox> currentSeq = new ArrayList<GenericGraphBox>();
+      for(int i = 0; i < zone.getSelectedBoxes().size(); i++) {
+        currentSeq.add(zone.getSelectedBoxes().get(i));
+      }
+      if (replaceSeq(zone, replaceSeqList, currentSeq)) {
+        res++;
+      }
+      zone.unSelectAllBoxes();
+    }
+    return res;
+  }
+
+  /**
+   * Replace seq with replaceSeqList and returns true if and only if the sequence has been replaced.
+   *
+   * @param zone the GraphicalZone containing the sequence.
+   * @param replaceSeqList the list containing the sequence to replace with.
+   * @param seq the list containing the sequence of boxes.
+     * @return true if the sequence has been replaced.
+     */
+  public static boolean replaceSeq(GraphicalZone zone, ArrayList<String> replaceSeqList, ArrayList<GenericGraphBox> seq) {
```

Comment:
I would combine the 4 if conditions into a single if else ladder because the conditions are mutually exclusive